aptos-move/framework/aptos-framework/sources/jwks.move

```diff
@@ -0,0 +1,480 @@
+        /// The utf-8 encoding of the issuer string (e.g., "https://accounts.google.com").
+        issuer: vector<u8>,
+
+        /// The `jwks` should each have a unique `id`, and should be sorted by `id` in alphabetical order.
+        jwks: vector<JWK>,
+    }
+
+    /// All OIDC providers and their JWK sets.
+    struct JWKMap has copy, drop, store {
+        /// Entries should each have a unique `issuer`, and should be sorted by `issuer` in an alphabetical order.
+        entries: vector<ProviderJWKSet>,
+    }
+
+    /// A `JWKMap` maintained by JWK consensus.
+    struct OnChainJWKMap has copy, drop, key, store {
+        version: u64,
+        jwk_map: JWKMap,
+    }
+
+    #[event]
+    /// When an on-chain JWK set update is done, this event is sent to reset the JWK consensus state in all validators.
+    struct OnChainJWKMapUpdated has drop, store {
+        epoch: u64,
+        version: u64,
+        jwk_map: JWKMap,
+    }
+
+    /// A small edit that can be applied to a `JWKMap`.
+    ///
+    /// Currently supported `content` types:
+    /// - `JWKMapEditCmdDelAll`
+    /// - `JWKMapEditCmdDelIssuer`
+    /// - `JWKMapEditCmdDelJwk`
+    /// - `JWKMapEditCmdPutJwk`
+    struct JWKMapEdit has drop, store {
+        content: Any,
+    }
+
+    struct JWKMapEditCmdDelAll has copy, drop, store {}
+
+    struct JWKMapEditCmdDelIssuer has copy, drop, store {
+        issuer: vector<u8>,
+    }
+
+    struct JWKMapEditCmdDelJwk has copy, drop, store {
+        issuer: vector<u8>,
+        jwk_id: vector<u8>,
+    }
+
+    struct JWKMapEditCmdPutJwk has copy, drop, store {
+        issuer: vector<u8>,
+        jwk: JWK,
+    }
+
+    /// A sequence of `JWKMapEdit` that needs to be applied *one by one* to the JWK consensus-maintained `JWKMap` before being used.
+    ///
+    /// Maintained by governance proposals.
+    struct JWKMapPatch has key {
+        edits: vector<JWKMapEdit>,
+    }
+
+    //
+    // Structs end.
+    // Functions begin.
+    //
+
+    /// Initialize some JWK resources. Should only be invoked by genesis.
+    public(friend) fun initialize(account: &signer) {
+        system_addresses::assert_aptos_framework(account);
+        move_to(account, OIDCProviderSet { providers: vector[] });
+        move_to(account, jwk_consensus_config_v0(10000));
+        move_to(account, OnChainJWKMap { version: 0, jwk_map: JWKMap { entries: vector[] } });
+        move_to(account, JWKMapPatch { edits: vector[] });
+    }
+
+    /// (1) Remove the entry for a provider of a given name from the provider set, if it exists.
+    /// (2) Add a new entry for the provider with the new config endpoint, if provided.
+    /// (3) Return the removed config endpoint in (1), if it happened.
+    ///
+    /// Designed to be used only in governance proposal-only.
+    public fun update_oidc_provider(account: &signer, name: vector<u8>, new_config_url: Option<vector<u8>>): Option<vector<u8>> acquires OIDCProviderSet {
```

Comment:
Wouldn't it be easier/clearer to have two different `remove_oidc_provider` and `add_oidc_provider` functions?

Also, don't understand how (2) makes sense: how can you ever add a provider without a config URL?